applications/plugins/LMConstraint/CMakeLists.txt

```diff
@@ -0,0 +1,83 @@
+cmake_minimum_required(VERSION 3.12)
+project(LMConstraint VERSION 1.0 LANGUAGES CXX)
+
+find_package(SofaFramework REQUIRED)
+find_package(SofaBase REQUIRED)
+find_package(SofaCommon REQUIRED) 
+find_package(SofaConstraint REQUIRED) 
+find_package(SofaDistanceGrid QUIET) 
+
+# List all files
+set(LMCONSTRAINT_SRC_DIR src/LMConstraint)
+set(HEADER_FILES
+    ${LMCONSTRAINT_SRC_DIR}/config.h.in
+    ${LMCONSTRAINT_SRC_DIR}/BaseLMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraint.inl
+    ${LMCONSTRAINT_SRC_DIR}/ContactDescription.h
+    ${LMCONSTRAINT_SRC_DIR}/DOFBlockerLMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/DOFBlockerLMConstraint.inl
+    ${LMCONSTRAINT_SRC_DIR}/FixedLMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/FixedLMConstraint.inl
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMContactConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMContactConstraint.inl
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMConstraint.inl
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraintSolver.h
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraintDirectSolver.h
+    ${LMCONSTRAINT_SRC_DIR}/MechanicalWriteLMConstraint.h
+    ${LMCONSTRAINT_SRC_DIR}/PrecomputedLMConstraintCorrection.h
+    ${LMCONSTRAINT_SRC_DIR}/PrecomputedLMConstraintCorrection.inl
+    )
+set(HEADER_FILES_TO_MOC
+    )
+set(SOURCE_FILES
+    ${LMCONSTRAINT_SRC_DIR}/initLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/BaseLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/DOFBlockerLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/FixedLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMContactConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/DistanceLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraintSolver.cpp
+    ${LMCONSTRAINT_SRC_DIR}/LMConstraintDirectSolver.cpp
+    ${LMCONSTRAINT_SRC_DIR}/MechanicalWriteLMConstraint.cpp
+    ${LMCONSTRAINT_SRC_DIR}/PrecomputedLMConstraintCorrection.cpp
+    )
+
+
+if(SofaDistanceGrid_FOUND)
+    list(APPEND HEADER_FILES
+
+        ${LMCONSTRAINT_SRC_DIR}/BarycentricDistanceLMConstraintContact.h
+        ${LMCONSTRAINT_SRC_DIR}/BarycentricDistanceLMConstraintContact.inl
+    )
+    list(APPEND SOURCE_FILES
+
+        ${LMCONSTRAINT_SRC_DIR}/BarycentricDistanceLMConstraintContact_DistanceGrid.cpp
+        ${LMCONSTRAINT_SRC_DIR}/BarycentricDistanceLMConstraintContact.cpp
+        ${LMCONSTRAINT_SRC_DIR}/TetrahedronBarycentricDistanceLMConstraintContact.cpp
+
+    )
+else()
+    message(STATUS "SofaDistanceGrid not found: DistanceLM codes will not be compiled")
+endif()
+
+
+# Create the plugin library
+add_library(${PROJECT_NAME} SHARED ${HEADER_FILES_TO_MOC} ${MOCCED_HEADER_FILES} ${HEADER_FILES} ${SOURCE_FILES} ${README_FILES})
```

Comment:
MOC thing comes from Qt so I guess you can remove everything related to it (especially because those variables does not contain anything in this file  😗 )